client/src/pages/Configurations/ConfigBuild/index.tsx

```diff
@@ -291,6 +436,147 @@ function ConditionCodeRow({
   );
 }
 
+interface AddCustomCodeModalProps {
+  open: boolean;
+  onClose: () => void;
+  onSubmit: (data: { code: string; system: string; name: string }) => void;
+  initialData?: { code: string; system: string; name: string } | null;
+}
+
+export function AddCustomCodeModal({
+  open,
+  onClose,
+  onSubmit,
+  initialData,
+}: AddCustomCodeModalProps) {
+  const modalRef = useRef<ModalRef>(null);
+  const showToast = useToast();
+
+  const [code, setCode] = useState(initialData?.code ?? '');
+  const [system, setSystem] = useState(initialData?.system ?? '');
+  const [name, setName] = useState(initialData?.name ?? '');
+
+  useEffect(() => {
+    if (open) {
+      setCode(initialData?.code ?? '');
+      setSystem(initialData?.system ?? '');
+      setName(initialData?.name ?? '');
+      modalRef.current?.toggleModal(undefined, true);
+    } else {
+      modalRef.current?.toggleModal(undefined, false);
+    }
+  }, [open, initialData]);
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    onSubmit({ code, system, name });
+
+    showToast({
+      heading: initialData ? 'Custom code updated' : 'Custom code added',
+      body: `${code}`,
+    });
+
+    onClose();
+  };
+
+  return (
+    <Modal
+      ref={modalRef}
+      id="add-custom-code-modal"
+      aria-labelledby="add-custom-code-title"
+      aria-describedby="Modal for adding custom code"
+      className="!h-[500px] !w-[35rem] rounded shadow"
```

Comment:
We should probably stick with `rem`. It also looks like [we don't want rounding](https://www.figma.com/design/QK7dVoNaG6qIF52r1imkwY/eCR-Refiner?node-id=2083-23841&m=dev).
```suggestion
      className="!h-[30.125rem] !w-[35rem] !rounded-none"
```